rasa/nlu/components.py

```diff
@@ -51,17 +51,46 @@ def validate_requirements(component_names: List[Text]) -> None:
         )
 
 
+def validate_tokenizers(pipeline: List["Component"]) -> None:
+    """Validates that only one tokenizer is present in the pipeline."""
+
+    from rasa.nlu.tokenizers.tokenizer import Tokenizer
+
+    tokenizer_names = []
+    for component in pipeline:
+        if isinstance(component, Tokenizer):
+            tokenizer_names.append(component.name)
+
+    if len(tokenizer_names) > 1:
+        raise InvalidConfigError(
+            f"More then one tokenizer is used: {tokenizer_names}. "
+            f"You can use only one tokenizer."
+        )
+
+
+def validate_required_components(pipeline: List["Component"]) -> None:
```

Comment:
I'll bring it up on this one again since the other PR got merged.

Here is why I don't think it is a good idea to require other components by name: it prevents other people from modifying (as in inheriting) from our components and adding them to the pipeline as a standin for the original component. This is why we have been matching on the ouput of other components using `requires`, allowing users to provide this input for a component however they like. 

Why is requiring `tokens` not enough in the example you posted on the other PR?

Reply:
`ConveRT` and all the other language models require a specific way of tokenization. They add special characters and split words in different ways into tokens. If we would use a different way of tokenization we may end up with a different number of tokens and features for a text as you need to pass the complete text into the language models in order to get the word embeddings. If the number of tokens and features differs the `DIETClassifier` would fail. We need to make sure that the tokenization and featurization matches.

Reply:
even spacy featurizer shouldn't be used with anything other then spacy tokenizer, because they create sequence features and tokens in parallel from spacy doc. We could specify `required_components` by type, then inheritance would be maintained

Reply:
Ok that makes sense. But in that case I think it makes sense to move away from `requires` as having both doesn't seem usefull.

How about:
- keeping only `requires`, adding `Component` classes to it, if it is a string we do the current check while at the same time deprecating the string format? 
- or alternatively keep `required_components` and deprecated `requires`

Reply:
then for classes like `CountVectorsFeaturizer` we would simply require any subclass of tokenizer, instead of message having tokens attribute?

Reply:
I it is also better for users, because they don't really need to know that a component needs certain message attributes

Reply:
@Ghostvv Can you take this over?

Reply:
yes

Reply:
https://github.com/RasaHQ/rasa/pull/5293